### ISSUE TITLE
componentWillReceiveProps warning in React 16.9

### ISSUE THREAD

**Issue by tonix-tuft** [wwayne/react-tooltip#507]
I just wanted to point out that I updated to `react@16.9.0` and `react-dom@16.9.0` and got the following warning in the console:

```
react-dom.development.js:11494 Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-async-component-lifecycle-hooks for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: ReactTooltip
```
![Screen Shot 2019-08-11 at 20 15 13](https://user-images.githubusercontent.com/10134421/62837871-e1adc200-bc74-11e9-9bfa-bd641347bd82.png)

It seems that `componentWillReceiveProps` will not work in the next major version of React.

**Comment by gfx** [comment]
refs: https://github.com/wwayne/react-tooltip/pull/487

**Comment by almilo** [comment]
+1

**Comment by julianCast** [comment]
This should be closed, fixed on #487
